src/execution_plan/optimizations/utilize_indices.c

```diff
@@ -28,12 +29,25 @@ int _reverseOp(int op) {
  * is of type variadic and the right-hand side is constant. */
 void _normalize_filter(FT_FilterNode *filter) {
 	// Normalize, left hand side should be variadic, right hand side const.
-	if(filter->pred.rhs->operand.type == AR_EXP_VARIADIC) {
-		// Swap.
-		AR_ExpNode *tmp = filter->pred.rhs;
-		filter->pred.rhs = filter->pred.lhs;
-		filter->pred.lhs = tmp;
-		filter->pred.op = _reverseOp(filter->pred.op);
+	switch(filter->t) {
+	case FT_N_PRED: {
+		if(filter->pred.rhs->operand.type == AR_EXP_VARIADIC) {
+			// Swap.
+			AR_ExpNode *tmp = filter->pred.rhs;
+			filter->pred.rhs = filter->pred.lhs;
+			filter->pred.lhs = tmp;
+			filter->pred.op = _reverseOp(filter->pred.op);
+		}
+		break;
+	}
+	case FT_N_COND: {
```

Comment:
I'm not sure if it is possible for `filter` to be of type `FT_N_COND` at this point,
have you seen this case being hit?

Reply:
removed